server/channels/app/authentication.go

```diff
@@ -61,6 +61,64 @@ func (a *App) IsPasswordValid(rctx request.CTX, password string) *model.AppError
 	return nil
 }
 
+// checkUserPassword tries to match the provided password against the user's
+// stored hash, migrating the latter if needed to the latest hashing method.
+func (a *App) checkUserPassword(user *model.User, password string, invalidateCache bool) *model.AppError {
+	err := users.CheckUserPassword(user, password)
```

Comment:
nit: Should we inline `users.CheckUserPassword` here? I found it surprising to have a second "check user password" method that does something relatively different than this one.

Reply:
I must admit I found it weird at first as well, then got used to it because of the different packages. But 2 people (100% of people that have read this) feeling weird about it is enough to revisit. Thanks!